app/models/user.rb

```diff
@@ -28,4 +28,20 @@ def member_of?(organisation)
   def application_names
     memberships.map(&:application_names).flatten.uniq
   end
+
+  def roles_for_application(application_id)
+    memberships
+      .joins(:application_memberships)
+      .where(application_memberships: {application_id: application_id})
+      .map{|m| m.roles_for_application(application_id)}
+      .flatten.uniq
+  end
+
+  def self.current
+    Thread.current[:user]
+  end
+
+  def self.current=(user)
+    Thread.current[:user] = user
+  end
```

Comment:
Just a word of warning that this will blow up in development mode when the class is reloaded as the instance will be using an old copy of the class. There is a `before_remove_const` hook that `ActiveSupport::Dependencies` calls before unloading the constant to give the app a chance to clean up after itself. You can see how it's used on the [Site model][1] for the new Petitions website.

[1]: https://github.com/alphagov/e-petitions/blob/master/app/models/site.rb#L15-L17